plugins/org.eclipse.fordiac.ide.util/src/org/eclipse/fordiac/util/marker/UtilityMarkerHelper.java

```
@@ -0,0 +1,123 @@
+		if (EditorUtils.getCurrentActiveEditor()
+				.getAdapter(LibraryElement.class) instanceof final AutomationSystem sys) {
+			try {
+				return Stream
+						.of(sys.getTypeEntry().getFile().findMarkers(markerDescriptor.ID(), true, IResource.DEPTH_ZERO))
+						.findAny().orElse(null);
+			} catch (final CoreException e) {
+				FordiacLogHelper.logError("Error accessing marked element", e); //$NON-NLS-1$
+			}
+		}
+		return null;
+	}
+
+	public static void deleteElementMarker(final MarkerDescriptor markerDescriptor, final IResource resource) {
+		deleteMarker(markerDescriptor.ID(), resource);
+	}
+
+	public static void deleteElementMarker(final String id) {
+		if ((EditorUtils.getCurrentActiveEditor()
+				.getAdapter(LibraryElement.class) instanceof final AutomationSystem sys)
+				&& (sys.getTypeEntry().getFile() instanceof final IResource resource)) {
+			deleteMarker(id, resource);
+		}
+	}
+
+	private static void deleteMarker(final String id, final IResource resource) {
+		try {
+			resource.deleteMarkers(id, false, IResource.DEPTH_ZERO);
+		} catch (final CoreException e) {
+			FordiacLogHelper.logError("Error deleting marker", e); //$NON-NLS-1$
+		}
+	}
+
+	public static void addElementMarker(final MarkerDescriptor markerDescriptor, final IResource resource,
+			final EObject target) {
+		try {
+			ErrorMarkerBuilder.createErrorMarkerBuilder(markerDescriptor.name()).setType(markerDescriptor.ID())
+					.setTarget(target).setSeverity(IMarker.SEVERITY_INFO).createMarker(resource);
+		} catch (final CoreException e) {
+			FordiacLogHelper.logError("Cannot add marker", e); //$NON-NLS-1$
+		}
+	}
+
+	public static boolean hasMarker(final MarkerDescriptor markerDescriptor, final IResource resource) {
+		try {
+			return Stream.of(resource.findMarkers(markerDescriptor.ID(), true, IResource.DEPTH_ZERO)).findAny()
+					.isPresent();
+		} catch (final CoreException e) {
+			FordiacLogHelper.logError("Cannot fetch marker", e); //$NON-NLS-1$
```

I would suggest to just silently return false here.